packages/agents-copilotstudio-client/src/copilotStudioClient.ts

```diff
@@ -169,13 +170,43 @@ export class CopilotStudioClient {
     yield * this.postRequestAsync(uriStart, body, 'POST')
   }
 
+  /**
+   * Sends an activity to the Copilot Studio service and retrieves the response activities.
+   * @param activity The activity to send.
+   * @param conversationId The ID of the conversation. Defaults to the current conversation ID.
+   * @returns An async generator yielding the Agent's Activities.
+   */
+  public async * sendActivityStreaming (activity: Activity, conversationId: string = this.conversationId) : AsyncGenerator<Activity> {
+    const localConversationId = activity.conversation?.id ?? conversationId
+    const uriExecute = getCopilotStudioConnectionUrl(this.settings, localConversationId)
+    const qbody: ExecuteTurnRequest = new ExecuteTurnRequest(activity)
+
+    logger.info('Sending activity...', activity)
+    yield * this.postRequestAsync(uriExecute, qbody, 'POST')
+  }
+
+  /**
+   * Starts a new conversation with the Copilot Studio service.
+   * @param emitStartConversationEvent Whether to emit a start conversation event. Defaults to true.
+   * @returns A promise yielding an array of activities.
+   * @deprecated Use startConversationStreaming instead.
+   */
+  public async startConversationAsync (emitStartConversationEvent: boolean = true): Promise<Activity[]> {
+    const result: Activity[] = []
+    for await (const value of this.startConversationStreaming(emitStartConversationEvent)) {
+      result.push(value)
+    }
+    return result
+  }
+
   /**
    * Sends a question to the Copilot Studio service and retrieves the response activities.
    * @param question The question to ask.
    * @param conversationId The ID of the conversation. Defaults to the current conversation ID.
-   * @returns An async generator yielding the Agent's Activities.
+   * @returns A promise yielding an array of activities.
+   * @deprecated Use sendActivityStreaming instead.
    */
-  public async * askQuestionAsync (question: string, conversationId: string = this.conversationId) : AsyncGenerator<Activity> {
+  public async askQuestionAsync (question: string, conversationId: string = this.conversationId) : Promise<Activity[]> {
```

Comment:
This looks like breaking changes,  going from streaming to non-streaming with the same sig. 
is this a bug fix or behavior change?